src/libs/actions/connections/QuickbooksOnline.ts

```diff
@@ -249,4 +405,169 @@ function updateQuickbooksOnlineReimbursableExpensesAccount<TConnectionName exten
+                            },
+                            errorFields: {
+                                [CONST.QUICK_BOOKS_CONFIG.REIMBURSEMENT_ACCOUNT_ID]: null,
+                            },
+                        },
+                    },
+                },
+            },
+        },
+    ];
+
+    const parameters: UpdateQuickbooksOnlineGenericTypeParams = {
+        policyID,
+        settingValue,
+        idempotencyKey: String(CONST.QUICK_BOOKS_CONFIG.REIMBURSEMENT_ACCOUNT_ID),
+    };
+    API.write(WRITE_COMMANDS.UPDATE_QUICKBOOKS_ONLINE_REIMBURSEMENT_ACCOUNT_ID, parameters, {optimisticData, failureData, successData});
+}
+
+function updateQuickbooksOnlinePreferredExporter(policyID: string, currentValue: string | undefined, settingValue: string) {
+    const optimisticData: OnyxUpdate[] = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
+            value: {
+                connections: {
+                    [CONST.POLICY.CONNECTIONS.NAME.QBO]: {
+                        config: {
+                            export: {
+                                exporter: settingValue ?? null,
+                            },
+                            pendingFields: {
+                                [CONST.QUICK_BOOKS_CONFIG.EXPORT]: CONST.RED_BRICK_ROAD_PENDING_ACTION.UPDATE,
+                            },
+                            errorFields: {
+                                exporter: null,
+                            },
+                        },
+                    },
+                },
+            },
+        },
+    ];
+
+    const failureData: OnyxUpdate[] = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
+            value: {
+                connections: {
+                    [CONST.POLICY.CONNECTIONS.NAME.QBO]: {
+                        config: {
+                            export: {
+                                exporter: currentValue ?? null,
+                            },
+                            pendingFields: {
+                                [CONST.QUICK_BOOKS_CONFIG.EXPORT]: null,
+                            },
+                            errorFields: {
+                                exporter: ErrorUtils.getMicroSecondOnyxErrorWithTranslationKey('common.genericErrorMessage'),
+                            },
+                        },
+                    },
+                },
+            },
+        },
+    ];
+
+    const successData: OnyxUpdate[] = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
+            value: {
+                connections: {
+                    [CONST.POLICY.CONNECTIONS.NAME.QBO]: {
+                        config: {
+                            export: {
+                                exporter: settingValue ?? null,
```

Comment:
same here. Can `settingValue` be null? 

If `settingValue` is null / undefined. I think we can do an early return and not make the API call at all.